bindings/java/src/main/java/org/apache/opendal/Environment.java

```diff
@@ -60,9 +62,35 @@ public enum Environment {
         } else {
             classifier.append("x86_64");
         }
+        if (classifier.toString().startsWith("linux-") && isMusl(arch)) {
+            classifier.append("-musl");
+        }
         INSTANCE.classifier = classifier.toString();
     }
 
+    private static boolean isMusl(String osArch) {
+        final String loader = muslLoaderName(osArch);
+        if (loader == null) {
+            return false;
+        }
+        return Files.exists(Paths.get("/lib", loader)) || Files.exists(Paths.get("/usr/lib", loader));
```

Comment:
Did you test this line works?

I suspect that with bundled jar the file doesn't exist on `/lib` but in the JAR package.